### PR TITLE
Add some quick checklist for releases

### PR DESCRIPTION
**Pull request by matthiasclasen**
This is how I've done releases so far.

### CODE REVIEW

File: RELEASE_HOWTO.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ - git clean -fxd
+ - bump version in configure.ac
+ - add content to NEWS
+ - git commit -m <version>
+ - git push origin master
+ - ./autogen.sh --enable-docbook-docs 
+ - make all
+ - make distcheck
+ - git tag <version>
+ - git push origin refs/tags/<version>
```
Comment (mariospr) [inline RELEASE_HOWTO.md]: I think I'd do `git push <version>`, which should work fine too (unless there's a branch with the same name).

But another option (and this is how I make releases on Endless) that I kind of like a bit more is **atomically** pushing the branch and the tags at the same time: don't do the `git push origin master` before, and simply do here `git push origin master <version>`

File: RELEASE_HOWTO.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+ - git clean -fxd
+ - bump version in configure.ac
+ - add content to NEWS
+ - git commit -m <version>
+ - git push origin master
+ - ./autogen.sh --enable-docbook-docs 
+ - make all
+ - make distcheck
+ - git tag <version>
```
Comment (mariospr) [inline RELEASE_HOWTO.md]: @matthiasclasen What do you thinkg of signing the tags with  your GPG key using `gpg -a -s <version>`? No strong feelings, but I think it's a nice thing to do when possible.

File: RELEASE_HOWTO.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+ - git push origin refs/tags/<version>
+ - upload tarball to github as release
+ - edit release, copy NEWS section in
```
Comment (mariospr) [inline RELEASE_HOWTO.md]: truncated line?

**Comment by mariospr** [comment]
Looks good, but I have a couple of small suggestions + a genuine question about what seems to be a truncated line.

File: RELEASE_HOWTO.md (hunk 2 above)
Comment (matthiasclasen) [reply RELEASE_HOWTO.md]: Feel free to. I don't do gpg

File: RELEASE_HOWTO.md (hunk 3 above)
Comment (matthiasclasen) [reply RELEASE_HOWTO.md]: No, I wrote it like this :-(
I meant: copy the NEWS section for the current release into the github release description and fix up the markup

File: RELEASE_HOWTO.md (hunk 2 above)
Comment (mariospr) [reply RELEASE_HOWTO.md]: Ok, still good to know you're not opposed to that

**Pull request merged**